nvflare/app_common/executors/task_exchanger.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Optional
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.executor import Executor
+from nvflare.apis.fl_constant import FLContextKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import ReturnCode, Shareable, make_reply
+from nvflare.apis.signal import Signal
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.fuel.utils.constants import PipeChannelName
+from nvflare.fuel.utils.pipe.pipe import Message, Pipe
+from nvflare.fuel.utils.pipe.pipe_handler import PipeHandler
+from nvflare.fuel.utils.validation_utils import check_non_negative_int, check_positive_number, check_str
+from nvflare.security.logging import secure_format_exception
+
+
+class TaskExchanger(Executor):
```

Comment:
It will be like this:

LauncherExecutor extends TaskExchanger , so they are on the nvflare client side.

Our client api will use FlareAgent, they are on the 3rd party side (user custom code side)

Reply:
Yes, so this is the first PR.

That refactoring will come the next PR along with changes to api

It need to change both left side and right side for things to work.